compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -356,6 +358,9 @@ public void visit(BLangFunction funcNode) {
         if (isLambda) {
             return;
         }
+
+        validateParams(funcNode);
```

Comment:
What about the return type?

Reply:
Here the purpose of validating params is to check the status of deprecation of each parameter. The return type is not checked here because we don't want to check the deprecation for the return type.

Reply:
Got is, so if I have a function that `returns Foo` and `Foo` is deprecated I'll get a warning when returning a value, but not for the return type in the function signature?

Reply:
```
const string CONST3 = "CONST3";
const string CONST4 = "CONST4";

@deprecated
public type T1 CONST3|CONST4;

public function func() returns T1 {
}
```

That means since the T1 is deprecated, the warning should be printed in `returns T1`?

Reply:
IMO, logging a warning for `returns T1` will be useful. 

WDYT? Looping in @hasithaa too.

Reply:
Logging a warning for return type is more useful for API developers.

Reply:
Fix it here 7dfcccc